test/util/TestResourceCreator.go

```diff
@@ -294,20 +335,71 @@ func (r *TestResourceCreator) DeleteAllTestResources(resourceNamesToNotDelete ..
 	for _, resourceToDelete := range kubegresList.Items {
 
 		kubegresPvcList, err := r.resourceRetriever.GetKubegresPvcByKubegresName(resourceToDelete.Name)
-		r.DeleteResource(&resourceToDelete, resourceToDelete.Name)
+		withWatch, stopWatcher := r.DeleteResourceWithWatch(&resourceToDelete, resourceToDelete.Name, kubegresList)
+		if withWatch != nil {
+			objectDeleteCount++
+			watchEventsChannels = append(watchEventsChannels, withWatch)
+			stopWatchers = append(stopWatchers, stopWatcher)
+		}
 
 		if err == nil {
 			for _, pvcToDelete := range kubegresPvcList.Items {
-				r.DeleteResource(&pvcToDelete, pvcToDelete.Name)
+				resourceWithWatch, stopWatcher := r.DeleteResourceWithWatch(&pvcToDelete, pvcToDelete.Name, pvcList)
+				if resourceWithWatch != nil {
+					objectDeleteCount++
+					watchEventsChannels = append(watchEventsChannels, resourceWithWatch)
+					stopWatchers = append(stopWatchers, stopWatcher)
+				}
 			}
 		} else {
 			log.Println("No PVC found for kubegres resource '" + resourceToDelete.Name + "'")
 			continue
 		}
 	}
 
-	log.Println("Deleted all resources created during tests. Waiting for 30 seconds...")
-	time.Sleep(30 * time.Second)
```

Comment:
watching for delete events rather than doing this sleep shave ~4min from each test group and hopefully helps with flakiness